lib/compile_time.flow

```diff
@@ -0,0 +1,34 @@
+export {
+	// Normally all these functions are eliminated from program before passing it to backends.
+
+	// Speculative function, used to ask to compute a constant value during compilation.
```

Comment:
Why is this speculative? It should request a compile time evaluation, and if fail, the compilatino should stop. That is not speculative. Just write that this causes the code to evaluate at compile time, and check that the result is a constant value. What is considered a constant value?